src/configure/clients/ITemplateServiceClient.ts

```diff
@@ -8,4 +8,5 @@ export interface ITemplateServiceClient {
     getTemplateParameters(templateId: string): Promise<ExtendedPipelineTemplate>;
     getTemplateConfiguration(templateId: string, inputs: StringMap<string>): Promise<ExtendedPipelineTemplate>;
     getTemplateFile(templateId: string, fileName: string): Promise<{ id: string, content: string }[]>;
+    getResourceFilteredTemplates(language: string, deployTarget: string): Promise<TemplateInfo[]>;
```

Comment:
function name should be GetTemplatesInfoByFilter and provide all three args like this
getTemplatesInfoByFilter( languageFilter: string, deployTargetFilter?: string, buildTargetFilter?: string) : Promise<TemplateInfo[]>